adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -583,9 +655,22 @@ private boolean isBrokerWithPRTSupport(final Intent intent) {
      */
     public String getCurrentUser() {
         // authenticator is not used if there is not any user
-        Account[] accountList = mAcctManager.getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
-        if (accountList.length > 0) {
-            return accountList[0].name;
+        if (isBrokerAccountServiceSupported()) {
```

Comment:
I'm not sure i'm following. you should always opt-in to use broker and prompt user to grant the permission if it's missing (we'll send you an error back to let you know it's missing). 

---
In reply to: [102327617](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/734#discussion_r102327617) [](ancestors = 102327617)